pkg/rancher-desktop/main/tray.ts

```diff
@@ -130,18 +132,19 @@ export class Tray {
    * triggered, close the watcher and restart after a duration (one second).
    */
   private async watchForChanges() {
-    const abortController = new AbortController();
+    for (const watcher of this.kubeConfigWatchers) {
+      watcher.close();
+    }
+    this.kubeConfigWatchers = [];
```

Comment:
Ignore me, I'm just code golfing:
```suggestion
    this.kubeConfigWatchers.splice(0).map(w => w.close());
```

Reply:
I thing the original code is much easier to understand, so I don't think this should be applied.

I also don't see how this sets `kubeConfigWatchers` to `[]`.

Reply:
`splice` modifies `this` (so it takes all items).

But yeah, don't take it :D

Reply:
You can find your `perl` over =====================> there. 😛